lark/lexer.py

```diff
@@ -91,6 +92,12 @@ def __init__(self, name, pattern, priority=1):
 
     def __repr__(self):
         return '%s(%r, %r)' % (type(self).__name__, self.name, self.pattern)
+    
+    def user_repr(self):
+        if self.name.startswith('__'): # We represent a generated terminal
+            return self.pattern.raw or self.name
```

Comment:
Isn't this a little confusing? What if the raw pattern looks like a name?

Reply:
raw pattern will always be the full source code, e.g. `"<>" or ``/[A-Z]+/i`

Reply:
You set it as `raw=literal.value`

What if raw="LOOKS_LIKE_TERMINAL" ?

Reply:
What is your suggestion?

Reply:
Maybe `repr(raw)` should be good enough.

But I can't say for sure, users are very creative in being confused.

Reply:
That would be more confusing. Then one would get `'"<>"'` as the what is printed, e.g. the string is `'\'"<>"\''`

Reply:
or with your example `'"LOOKS_LIKE_TERMINAL"'`

Reply:
The quotes are already included.